pkg/util/cachepersister.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import "github.com/golang/glog"
+
+const (
+	PluginFolder = "/var/lib/kubelet/plugins"
+)
+
+type ForAllFunc func(identifier string) error
+
+type CachePersister interface {
+	Create(identifier string, data interface{}) error
+	Get(identifier string, data interface{}) error
+	ForAll(pattern string, destObj interface{}, f ForAllFunc) error
+	Delete(identifier string) error
+}
+
+func NewCachePersister(metadataStore string, driverName string) CachePersister {
+	if metadataStore == "k8s_configmap" {
+		glog.Infof("cache-perister: using kubernetes configmap as metadata cache persister")
+		k8scm := &K8sCMCache{}
+		k8scm.Client = NewK8sClient()
+		k8scm.Namespace = GetK8sNamespace()
+		return k8scm
+	} else {
```

Comment:
I'd prefer checking for equality here and returning an error in case of mismatch instead of silently using another cache storage (we don't want to be using node cache just because of a typo in the cmd args).